### ISSUE TITLE
Subscriber does not receive messages from subscribed Azure Event Hubs instance topic

### ISSUE THREAD

**Issue by james-stallard** [dapr/dapr#7123]
## In what area(s)?

/area runtime

## Ask your question here

I am trying to get Pub/Sub working with my Azure Event Hubs instance, but can not get my subscriber to receive messages.
Any advice on how to resolve my problem would be appreciated.

This is my subscriber implementation in Python:
```
import uvicorn
from dapr.ext.fastapi import DaprApp
from fastapi import Body, FastAPI

app = FastAPI()
dapr_app = DaprApp(app)


@dapr_app.subscribe(pubsub='pubsub', topic='daprtest')
def cloud_event_handler(event_data=Body()):
    print(event_data)


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=5000)
```

When executed along with my publisher, using Kafka, for example, it just works.

```
[dapr-pub] INFO:root:Published data: {"name": "Jane Doe"}
[dapr-sub] {'data': {'name': 'Jane Doe'}, 'datacontenttype': 'application/json', 'id': '6e261d6d-4313-4e0d-b4fe-5ce6dfb15a4a', 'pubsubname': 'pubsub', 'source': 'dapr-pub', 'specversion': '1.0', 'time': '2023-10-31T07:37:04Z', 'topic': 'daprtest', 'traceid': '00-00000000000000000000000000000000-0000000000000000-00', 'traceparent': '00-00000000000000000000000000000000-0000000000000000-00', 'tracestate': '', 'type': 'com.dapr.event.sent'}
```

However, when executed along with my publisher, using Azure Event Hubs, the subscriber subscribes, but does not receive any messages.

Logs from my subscriber, indicating that it successfully subscribed... but then receives nothing.

```
❯ k logs -f dapr-sub-6fb885d5c7-tpggwDefaulted container "dapr-sub" out of: dapr-sub, daprd
INFO:     Started server process [1]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:5000 (Press CTRL+C to quit)
INFO:     127.0.0.1:54468 - "GET /dapr/config HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:54474 - "GET /dapr/subscribe HTTP/1.1" 200 OK
```

Logs from my subscriber `daprd`.

```
...
time="2023-10-31T07:39:51.917444485Z" level=info msg="The provided connection string is specific to the Event Hub (\"entity path\") 'daprtest'; publishing or subscribing to a topic that does not match this Event Hub will fail when attempted" app_id=dapr-sub component="pubsub (pubsub.azure.eventhubs/v1)" instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.contrib type=log ver=1.12.0
time="2023-10-31T07:39:51.917542776Z" level=info msg="Component loaded: pubsub (pubsub.azure.eventhubs/v1)" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.917555393Z" level=info msg="All outstanding components processed" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.917576876Z" level=info msg="Loading endpoints…" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.917982527Z" level=info msg="Waiting for all outstanding http endpoints to be processed…" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.917998467Z" level=info msg="All outstanding http endpoints processed" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.918179915Z" level=error msg="Failed to listen for gRPC server on TCP address [::1]:50001 with error: listen tcp [::1]:50001: bind: cannot assign requested address" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.api type=log ver=1.12.0
time="2023-10-31T07:39:51.918260469Z" level=info msg="gRPC server listening on TCP address: 127.0.0.1:50001" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.api type=log ver=1.12.0
time="2023-10-31T07:39:51.918269708Z" level=info msg="Enabled gRPC tracing middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.api type=log ver=1.12.0
time="2023-10-31T07:39:51.918275595Z" level=info msg="Enabled gRPC metrics middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.api type=log ver=1.12.0
time="2023-10-31T07:39:51.918335286Z" level=info msg="Registering workflow engine for gRPC endpoint: 127.0.0.1:50001" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.api type=log ver=1.12.0
time="2023-10-31T07:39:51.918359103Z" level=info msg="API gRPC server is running on port 50001" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.918424999Z" level=info msg="Enabled max body size HTTP middleware with size 4 MB" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.918438124Z" level=info msg="Enabled tracing HTTP middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.918443466Z" level=info msg="Enabled metrics HTTP middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.918821222Z" level=info msg="HTTP server listening on TCP address: 127.0.0.1:3500" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.918919144Z" level=info msg="Enabled tracing HTTP middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.918982081Z" level=info msg="Enabled metrics HTTP middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.http type=log ver=1.12.0
time="2023-10-31T07:39:51.919027643Z" level=info msg="HTTP server is running on port 3500" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.919062464Z" level=info msg="The request body size parameter is: 4" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.919088036Z" level=info msg="gRPC server listening on TCP address: :50002" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.internal type=log ver=1.12.0
time="2023-10-31T07:39:51.919101718Z" level=info msg="Enabled gRPC tracing middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.internal type=log ver=1.12.0
time="2023-10-31T07:39:51.91910719Z" level=info msg="Enabled gRPC metrics middleware" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.grpc.internal type=log ver=1.12.0
time="2023-10-31T07:39:51.919168552Z" level=info msg="Internal gRPC server is running on port 50002" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:51.919208896Z" level=info msg="application protocol: http. waiting on port 5000.  This will block until the app is listening on that port." app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:52.321022242Z" level=info msg="application discovered on port 5000" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:52.322723931Z" level=info msg="Application configuration loaded" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:52.322885536Z" level=info msg="actors: state store is not configured - this is okay for clients but services with hosted actors will fail to initialize!" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:52.32298876Z" level=info msg="Actor runtime started. Actor idle timeout: 1h0m0s. Actor scan interval: 30s" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.actor type=log ver=1.12.0
time="2023-10-31T07:39:52.323026043Z" level=info msg="Configuring workflow engine with actors backend" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.wfengine type=log ver=1.12.0
time="2023-10-31T07:39:52.323149559Z" level=info msg="Registering component for dapr workflow engine..." app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:52.323199838Z" level=info msg="Initializing Dapr workflow component" app_id=dapr-sub component="dapr (workflow.dapr/v1)" instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.contrib type=log ver=1.12.0
time="2023-10-31T07:39:52.329510688Z" level=info msg="app is subscribed to the following topics: [daprtest] through pubsub=pubsub" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.processor.pubsub type=log ver=1.12.0
time="2023-10-31T07:39:52.424693778Z" level=info msg="Placement tables updated, version: 0" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime.actors.placement type=log ver=1.12.0
time="2023-10-31T07:39:52.459514384Z" level=info msg="dapr initialized. Status: Running. Init Elapsed 545ms" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.runtime type=log ver=1.12.0
time="2023-10-31T07:39:56.371980751Z" level=info msg="HTTP API Called" app_id=dapr-sub instance=dapr-sub-6fb885d5c7-tpggw method="GET /v1.0/healthz" scope=dapr.runtime.http-info type=log useragent=kube-probe/1.27ver=1.12.0
```

Specifically, these lines indicate that it has successfully subscribed to the `daprtest` topic.
```
time="2023-10-31T07:39:51.917444485Z" level=info msg="The provided connection string is specific to the Event Hub (\"entity path\") 'daprtest'; publishing or subscribing to a topic that does not match this Event Hub will fail when attempted" app_id=dapr-sub component="pubsub (pubsub.azure.eventhubs/v1)" instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.contrib type=log ver=1.12.0
...
time="2023-10-31T07:39:52.323199838Z" level=info msg="Initializing Dapr workflow component" app_id=dapr-sub component="dapr (workflow.dapr/v1)" instance=dapr-sub-6fb885d5c7-tpggw scope=dapr.contrib type=log ver=1.12.0
time="2023-10-31T07:39:52.329510688Z" level=info msg="app is subscribed to the following topics: [daprtest] through pubsub=pubsub"
```

This is my Azure Event Hubs Component manifest:

```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: pubsub
spec:
  type: pubsub.azure.eventhubs
  version: v1
  metadata:
    - name: connectionString
      value: 'Endpoint=sb://{redacted}.servicebus.windows.net/;SharedAccessKeyName=test;SharedAccessKey={redacted};EntityPath=daprtest'
    - name: enableEntityManagement
      value: "false"
    - name: storageAccountName
      value: "{redacted}"
    - name: storageAccountKey
      value: "{redacted}"
    - name: storageContainerName
      value: "test"
```

**Comment by james-stallard** [comment]
I was able to resolve the issue.

Apparently my mistake was that I missed this portion of the [instructions](https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-eventhubs/#create-consumer-groups-for-each-subscriber).

> For every Dapr app that wants to subscribe to events, create an Event Hubs consumer group with the name of the Dapr app ID. For example, a Dapr app running on Kubernetes with dapr.io/app-id: "myapp" will need an Event Hubs consumer group named myapp.

I changed the name of the Azure Storage Container name and consumer group to be identical to what is set in `dapr.io/app-id`, and the subscriber started to work.

Hope someone finds this comment useful.

I will close this issue now.
